taskrun/TaskManager.py

```diff
@@ -223,6 +223,25 @@ def task_failed(self, task, errors):
     This function is called by a task when an error code is returned from
     the task
 
+    Args:
+      task (Task) : the task that encountered errors
+    """
+    self._task_failed_or_killed(task, errors)
+
+  def task_killed(self, task):
+    """
+    This function is called by a task when it has been killed
+
+    Args:
+      task (Task) : the task that was killed
+    """
+    self._task_failed_or_killed(task, None)
+
+  def _task_failed_or_killed(self, task, errors):
+    """
+    This function is called by a task when an error code is returned from
+    the task
```

Comment:
Not sure that "when an error code is returned from the task" is accurate. Maybe when a task did not complete successfully?